### PR TITLE
Phone Update part 2,  Call History, Delete Call History, Silence mode, Add Published contacts as contacts toggle.

### PR DESCRIPTION
**Pull request by wgtjunior743**
## About The Pull Request

Now you can see all your call history, from those who called you or if you called them , see in the pictures.
You can silence your phone and get zero noise, also toggle a new settings to auto add all published contacts to your contact list.

New button too, focused on settings, that was my first time touching tgui and react, so it ain't perfect, if someone with a better knowledge on that and wants to beautify , feel free to do it please.

Ofcourse it is compiled tested, i tested it dozens of times as well but it would be good to have another pair of eyes.

The new settings button.
new_phone

The new options (this version i was working don't had the block, unblock etc)
History_Del_History

The toggles:
sound_notifi

Some examples for the call history:
call_history_2
call_history_1

## Why It's Good For The Game

More stuff for phones, better phones.

## Changelog
:cl:
add: new settings button, changed abit of the tgui telephone code
add: new silence toggle at the settings button
add: new Toggle Published numbers as contacts at the settings button
add: new Call History and Delete History for the contact options
/:cl:

### CODE REVIEW

**Review by lucifernix — CHANGES_REQUESTED** [review]
phone_history.dm is not in tgstation.dme, it was never ticked on to be active in dreammaker.

Also, I get the following errors:

code\modules\vtmb\electronics\phones\phone.dm:558:error: ADDED_CONTACTS: undefined var
code\modules\vtmb\electronics\phones\phone.dm:559:error: ADDED_CONTACTS: undefined var
code\modules\vtmb\electronics\phones\phone.dm:345:error: else: 'else' clause without preceding 'if' statement

I was unable to get it to compile.

**Comment by wgtjunior743** [comment]
> phone_history.dm is not in tgstation.dme, it was never ticked on to be active in dreammaker.
> 
> Also, I get the following errors:
> 
> code\modules\vtmb\electronics\phones\phone.dm:558:error: ADDED_CONTACTS: undefined var code\modules\vtmb\electronics\phones\phone.dm:559:error: ADDED_CONTACTS: undefined var code\modules\vtmb\electronics\phones\phone.dm:345:error: else: 'else' clause without preceding 'if' statement
> 
> I was unable to get it to compile.

fixed that on the news commits, somehow i fucked up the identation when i was transfering the code from DM to github and ofcourse forgot one var, the new dm is added the tgstation.dme, forgot you had to do it manually here

**Comment by wgtjunior743** [comment]
> Hmm. I am still getting some warnings when trying to compile.
> 
> image image

fixed on the new commit, it does the exactly same thing, just better byond wise and more friendly, to say the truth i had no idea you could run a for like that in byond, what i tried using was closer to java.
The warning is because before it was not searching for a specific datum but variables with those names on the list, it worked because it was not being used for any other stuff besides the phonehistory datum so ofcourse it would have that variables, but byond don't knew that and was saying "specify this type because it could have or not what you are asking for", that warning also does not show on DM for some ungodly reason.

File: code/modules/vtmb/electronics/phones/phone.dm [hunk 1]
```diff
@@ -365,13 +480,92 @@
...
+										ADDED_CONTACTS +=1
+							if(ADDED_CONTACTS > 1)
+								to_chat(usr, "<span class='notice'>New contacts are added to your contact list.</span>")
+						else if(contacts_added_lenght == list_length)
+							to_chat(usr, "<span class='notice'>You have all the contacts in the published list already.</span>")
+						toggle_published_contacts = TRUE
+						to_chat(usr, "<span class='notice'>The toggle of the published numbers in contacts is active.</span>")
+					else
+						toggle_published_contacts = FALSE
+						to_chat(usr, "<span class='notice'>The toggle of the published numbers in contacts is desactive.</span>")
```
Comment (lucifernix) [inline code/modules/vtmb/electronics/phones/phone.dm:564]:
```suggestion
						to_chat(usr, "<span class='notice'>The toggle of the published numbers in contacts is disabled.</span>")
```

**Pull request merged**